zokrates_core/src/flatten/mod.rs

```diff
@@ -11,19 +11,18 @@ use crate::typed_absy::*;
 use crate::types::conversions::cast;
 use crate::types::Signature;
 use crate::types::Type;
-use bimap::BiMap;
 use std::collections::HashMap;
 use zokrates_field::field::Field;
 
 /// Flattener, computes flattened program.
 #[derive(Debug)]
-pub struct Flattener {
+pub struct Flattener<'ast> {
     /// Index of the next introduced variable while processing the program.
     next_var_idx: usize,
     ///
-    bijection: BiMap<String, FlatVariable>,
+    layout: HashMap<Identifier<'ast>, Vec<FlatVariable>>,
```

Comment:
Any Identifier in the TypedProgram maps to a series of FlatVariable (one for Field, n for FieldArray(n))